lib/aemo/nem12.rb

```diff
@@ -349,15 +349,19 @@ def parse_nem12_200(line, options = {})
     # @return [Array of hashes] the line parsed into a hash of information
     def parse_nem12_300(line, options = {})
       csv = line.parse_csv
-
       raise TypeError, 'Expected NMI Data Details to exist with IntervalLength specified' if @data_details.last.nil? || @data_details.last[:interval_length].nil?
+
+      # ref: AEMO's MDFF Spec NEM12 and NEM13 v1.01 (2014-05-14)
+      record_fixed_fields = %w[RecordIndicator IntervalDate QualityMethod ReasonCode ReasonDescription UpdateDatetime MSATSLoadDateTime]
       number_of_intervals = 1440 / @data_details.last[:interval_length]
+      raise TypeError, 'Invalid record length' if csv.length != record_fixed_fields.length + number_of_intervals
+
       intervals_offset = number_of_intervals + 2
 
       raise ArgumentError, 'RecordIndicator is not 300' if csv[0] != '300'
       raise ArgumentError, 'IntervalDate is not valid' if csv[1].match(/\d{8}/).nil? || csv[1] != Time.parse(csv[1].to_s).strftime('%Y%m%d')
       (2..(number_of_intervals + 1)).each do |i|
-        raise ArgumentError, "Interval number #{i - 1} is not valid" if csv[i].match(/\d+(\.\d+)?/).nil?
+        raise ArgumentError, "Interval number #{i - 1} is not valid" if csv[i].nil? || csv[i].match(/\d+(\.\d+)?/).nil?
```

Comment:
Can we have a test for this too please?

Add a NEM12 file with some of the numbers set to empty "columns":

```csv
100,NEM12,200505181432,CNRGYMDP,NEMMCO
200,NEM1201002,E1,E1,E1,N1,01002,KWH,30,
300,20050315,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,A,,,20050316014209,
900
```

Reply:
I have added test cases. Please see my recent changes.